web/client-api/src/test/java/io/deephaven/web/client/api/grpc/GrpcTransportGwtTest.java

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.web.client.api.grpc;
+
+import elemental2.promise.Promise;
+import io.deephaven.web.client.api.AbstractAsyncGwtTestCase;
+import io.deephaven.web.client.api.ConnectOptions;
+import io.deephaven.web.client.api.CoreClient;
+import jsinterop.base.JsPropertyMap;
+
+/**
+ * Simple test to verify we can produce custom transports in JS. Only works with https, which means it can only be run
+ * manually at this time, or it will trivially succeed.
+ */
+public class GrpcTransportGwtTest extends AbstractAsyncGwtTestCase {
+    @Override
+    public String getModuleName() {
+        return "io.deephaven.web.DeephavenIntegrationTest";
+    }
+
+    /**
+     * Simple fetch impl, with no cancelation handling.
+     */
+    public native GrpcTransportFactory makeFetchTransportFactory() /*-{
+        return {
```

Comment:
Is the commented out section left intentionally?

Reply:
Not commented out - this is "JSNI", an extension to java that lets us write (a subset of) JS in java and have it left alone when compiled. That way I can write some plain JS and not have Java accidentally fill in the blanks when implementing the interfaces (like with that adapt() method that seemed missing when you implemented this in an earlier revision).

The `native` keyword signals that there is an implementation to this in plain JS somewhere, and the `/*-{...}-*/;` section is the inline implementation.